### ISSUE TITLE
SPA associative memory does not accept list of vectors

### ISSUE THREAD

**Issue by jgosmann** [nengo/nengo#1120]
The `AssociativeMemory` documentation states

``` rst
    input_vocab: list or Vocabulary
        The vocabulary (or list of vectors) to match.
    output_vocab: list or Vocabulary, optional (Default: None)
        The vocabulary (or list of vectors) to be produced for each match. If
        None, the associative memory will act like an autoassociative memory
        (cleanup memory).
```

However, using a list for `output_vocab` an `AttributeError` is raised:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-8-669c3990bc85> in <module>()
      8         model.cues = spa.State(d)
      9         model.proposed = spa.State(d)
---> 10     model.am = spa.AssociativeMemory(vocab, [x for x in np.dot(assoc_fang, vocab.vectors)])
     11     model.keep_response = spa.Compare(d, neurons_per_multiply=400)
     12 

/home/jgosmann/.virtualenvs/bleeding-edge/lib/python2.7/site-packages/nengo-2.1.1.dev0-py2.7.egg/nengo/spa/assoc_mem.pyc in __init__(self, input_vocab, output_vocab, input_keys, output_keys, default_output_key, threshold, inhibitable, wta_output, wta_inhibit_scale, wta_synapse, threshold_output, label, seed, add_to_container)
     71             if output_keys is None:
     72                 output_keys = input_keys
---> 73             output_vectors = output_vocab.create_subset(output_keys).vectors
     74 
     75         if default_output_key is None:

AttributeError: 'list' object has no attribute 'create_subset'
```

Either the `AssociativeMemory` should be fixed to accept lists or the documentation should be updated.

**Comment by xchoo** [comment]
Oh! I thought we fixed this. It's just an error in the documentation. For people who want to pass vectors to an associative memory, they'd want to use the `nengo.networks.AssociativeMemory` network.

**Comment by jgosmann** [comment]
While the support will likely be added in new-spa, we should
- [ ] correct the legacy SPA documentation.

**Comment by jgosmann** [comment]
@Seanny123 Why are you assigned to this? This is such a quick fix for legacy SPA that I'm probably going to make a PR in the next hour.

**Comment by xchoo** [comment]
Hmmm. While the documentation should be corrected, i still think there should be a difference between the `spa` and `networks` versions (i.e. one should work purely with vocabs, the other vectors, respectively)

**Comment by jgosmann** [comment]
The rewritten new-spa networks can be used as either a standard Nengo network or SPA module because I don't see a good reason to differentiate between the two in the new system. I'll aim to have a single network for the associative memory too, but I'm just starting to look into this, so I might come across something that makes separate networks better.

**Comment by tbekolay** [comment]
Docs fixed in #1215, thanks @jgosmann !
